src/sys/unix.rs

```diff
@@ -1597,6 +1597,93 @@ impl crate::Socket {
         ))
         .map(|_| sbytes as usize)
     }
+
+    /// Set the value of the `TCP_USER_TIMEOUT` option on this socket.
+    ///
+    /// If set, this specifies the maximum amount of time that transmitted data may remain
+    /// unacknowledged or buffered data may remain untransmitted before TCP will forcibly close the
+    /// corresponding connection.
+    #[cfg(all(
+        feature = "all",
+        any(
+            target_os = "android",
+            target_os = "emscripten",
+            target_os = "fuchsia",
+            target_os = "l4re",
+            target_os = "linux",
+        )
+    ))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(all(
+            feature = "all",
+            any(
+                target_os = "android",
+                target_os = "emscripten",
+                target_os = "fuchsia",
+                target_os = "l4re",
+                target_os = "linux",
+            )
+        )))
+    )]
+    pub fn set_tcp_user_timeout(&self, duration: Option<Duration>) -> io::Result<()> {
+        use std::convert::TryInto;
+        fn into_millis(duration: Option<Duration>) -> io::Result<c_int> {
```

Comment:
Per the `TCP_USER_TIMEOUT` docs at https://man7.org/linux/man-pages/man7/tcp.7.html:

> This option takes an unsigned int as an argument.

So this should be a `libc::c_uint`.

Reply:
Was resolved by switching to clamping instead of emitting an error on oversized duration